ansible/roles/distributed_press/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+
+# Vars are prefixed with distributed_press_
+
+- name: Ensure user exists for DP service
+  user:
+    name: "{{distributed_press_user}}"
+    append: true
+    # TODO: is this needed?
+    groups: adm
+    shell: /bin/bash
+    comment: User for the distributed.press API server
+
+- name: "Set up node+npm"
+  include_role:
+    name: geerlingguy.nodejs
+
+- name: Add distributed.press systemd service
+  template:
+    src: distributed.press.service.j2
+    dest: "/etc/systemd/system/{{distributed_press_service_name}}.service"
+
+- name: "Stop distributed.press service to recompile"
+  systemd:
+    state: stopped
+    name: "{{distributed_press_service_name}}"
+
+- name: Clone or update DP codebase
+  become: yes
+  become_user: "{{distributed_press_user}}"
+  git:
+    repo: "{{distributed_press_git_repo}}"
+    version: "{{distributed_press_git_branch}}"
+    dest: "{{distributed_press_source}}"
+    clone: yes
+    update: yes
+
+- name: Install NPM Dependencies
+  become: yes
+  become_user: "{{distributed_press_user}}"
+  shell: |
```

Comment:
We should add a few TODO comments that point out what needs to change when v1 is eventually merged (e.g. changing the branch)

As a separate question: do we eventually want to get rid of the v0 folder as well?

Reply:
Yeah, I feel like it'd be safe to get rid of v0 as part of merging v1 in.

Maybe in the README we could point to the last stable v0 release as a tag for those that need it (our legacy deploys I think).

Maybe we can get to that once we merge v1 in to the main branch?

Reply:
I think the key in the comment I think your referring to is 
> Typically applications should increase their soft limit to the hard limit on their own
Which I would assume the app does not do.

 I don't see the danger other then not using `select(2)` and possible resource utilization (DDOS). In un-managed code I can see leaks of not closing descriptors when your done with them, but with a GC I don't think its much of a danger if leakage.

I have seen this value set much higher in other apps esp distributed systems I have seen this value set much higher then 4096. For example in Saturn the docker container is set to 10,000 (but i guess that is system wide not for a single process)
https://github.com/filecoin-saturn/L1-node/blob/main/docker-compose.yml#L12-L15

Keep in mind this is the `Distributed Press API Server` its going to be touching lots of file descriptors, form incoming sockets, to outgoing connections to reading files and metadata.

Just my 2 cents.

Reply:
Gonna leave this a s a TODO for when we merge v1-staging into the main branch.